### PR TITLE
jupyter: import external unit metrics

### PR DESCRIPTION
**Pull request by magland**
Allow importing of external unit metrics from jupyter notebook as follows:

image

### CODE REVIEW

File: jupyterlab/labbox_ephys_widgets_jp/labbox_ephys_widgets_jp/create_sorting_view.py [hunk 1]
```diff
@@ -39,6 +40,7 @@ class SortingView(DOMWidget):
         sortingInfo = DictTrait(le.get_sorting_info(sorting_object=sorting.object(), recording_object=recording.object())).tag(sync=True)
         selection = DictTrait({}).tag(sync=True)
         curation = DictTrait({}).tag(sync=True)
+        externalUnitMetrics = ListTrait([]).tag(sync=True)
```
Comment (magland) [inline jupyterlab/labbox_ephys_widgets_jp/labbox_ephys_widgets_jp/create_sorting_view.py:43]: externalUnitMetrics is now an attribute of the sorting view in jupyter. You should be able to set it either before or after the widget is created.

File: jupyterlab/labbox_ephys_widgets_jp/package.json [hunk 2]
```diff
@@ -11,6 +11,8 @@
   "files": [
     "lib/**/*.js",
     "dist/*.js",
+    "lib/**/*.css",
```
Comment (magland) [inline jupyterlab/labbox_ephys_widgets_jp/package.json:14]: Had some trouble with the new css files during building of the jupyterlab extension

File: jupyterlab/labbox_ephys_widgets_jp/package.json [hunk 3]
```diff
@@ -31,7 +33,7 @@
   "scripts": {
     "build": "npm run build:lib && npm run build:nbextension",
     "build:labextension": "npm run clean:labextension && mkdirp labbox_ephys_widgets_jp/labextension && cd labbox_ephys_widgets_jp/labextension && npm pack ../..",
-    "build:lib": "tsc",
+    "build:lib": "tsc && npm run copy-files",
```
Comment (magland) [inline jupyterlab/labbox_ephys_widgets_jp/package.json:36]: Turns out that tsc only transpiles ts -> js and does not copy other files (such as .css) In order to handle css properly with the jupyterlab extension, we need to have a step that copies those css files into the lib/ directory. See the copy-files command below.

File: jupyterlab/labbox_ephys_widgets_jp/src/pluginWidgets.tsx [hunk 4]
```diff
@@ -249,8 +250,24 @@ const PluginComponentWrapper: FunctionComponent<PluginComponentWrapperProps> = (
       selection: model.get('selection')
     })
   }, null)
+
+  // externalUnitMetrics
```
Comment (magland) [inline jupyterlab/labbox_ephys_widgets_jp/src/pluginWidgets.tsx:254]: We handle the externalUnitMetrics attribute in a very similar way to curation and selection attributes.

File: src/extensions/extensionInterface.ts [hunk 5]
```diff
@@ -172,6 +172,20 @@ export const sortingCurationReducer: Reducer<SortingCuration, SortingCurationAct
 }
 ////////////////////
 
+// This reducer is used only by the jupyter extension
```
Comment (magland) [inline src/extensions/extensionInterface.ts:175]: A new reducer for use in the jupyter extension

**Review by jsoules — COMMENTED** [review]
All seems reasonable.

**Pull request merged**